kong/error_test.go

```diff
@@ -49,3 +51,36 @@ func TestAPIError_Code(T *testing.T) {
 	require.True(ok)
 	assert.True(kongErr.Code() == 404)
 }
+
+func TestIsForbiddenErrE2E(T *testing.T) {
+	assert := assert.New(T)
+
+	// Create the http client with a custom transport
+	// always returning a 403
+	forbiddenTransport := &forbiddenTransport{}
+	httpClient := &http.Client{
+		Transport: forbiddenTransport,
+	}
+	// if KONG_ADMIN_TOKEN is set, a different custom transport would be used
+	T.Setenv("KONG_ADMIN_TOKEN", "")
+	adminToken := os.Getenv("KONG_ADMIN_TOKEN")
+	os.Unsetenv("KONG_ADMIN_TOKEN")
+	defer func() {
+		os.Setenv("KONG_ADMIN_TOKEN", adminToken)
+	}()
```

Comment:
`t.Setenv()` will automatically restore the env to previous value using `t.Cleanup()`

```suggestion
	// if KONG_ADMIN_TOKEN is set, a different custom transport would be used
	T.Setenv("KONG_ADMIN_TOKEN", "")
```

Reply:
I tried that and it didn't work, but let's give it another try

Reply:
Yeah it doesn't work: https://github.com/Kong/go-kong/actions/runs/3910665439/jobs/6683126778

The `os.LookupEnv("KONG_ADMIN_TOKEN")` returns `true` even if the env value is set to an empty string. Unfortunately, golang testing lib doesn't have the equivalent of `Unsetenv` :\

Reply:
Let me know if something like this would be okay for you https://github.com/Kong/go-kong/pull/264/commits/102506c5fd1cfcad2ac9bc84ff2b78b991b24718

Reply:
If that works then sure 👍